### PR TITLE
[AIRFLOW-XXX] Correct BranchPythonOperator docs

### PR DESCRIPTION
**Pull request by dima-asana**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - This is a minor documentation change, so following ^^

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Here's the code showing that skipped tasks satisfy depends_on_past:
https://github.com/apache/airflow/blob/de75b7a2bd7f5bef6a1d09942e0b43c17a3fbb95/airflow/ti_deps/deps/prev_dagrun_dep.py#L75

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Docs only

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by OmerJog** [comment]
There is another pending PR https://github.com/apache/airflow/pull/4388 on the same docs rows
@BasPH @jgao54

File: airflow/operators/python_operator.py [hunk 1]
```diff
@@ -132,10 +132,8 @@ class BranchPythonOperator(PythonOperator, SkipMixin):
     to be inferred.
 
     Note that using tasks with ``depends_on_past=True`` downstream from
-    ``BranchPythonOperator`` is logically unsound as ``skipped`` status
-    will invariably lead to block tasks that depend on their past successes.
-    ``skipped`` states propagates where all directly upstream tasks are
-    ``skipped``.
+    ``BranchPythonOperator`` is fine because depends_on_past allows for
+    past tasks to be skipped.
```
Comment (ashb) [inline airflow/operators/python_operator.py]: Lets just remove the paragraph entirely if this is the case? (Since when was this the case?)

Comment (dima-asana) [reply airflow/operators/python_operator.py]: Done, sorry for the delay.

The behavior changed initially in https://github.com/apache/airflow/commit/2630361ca24737c28f458825b20ab11c9c996b17, leading to https://github.com/apache/airflow/commit/f360414774f1ecb2e0f4e53ebdd623c7435f9a78#diff-a32a363fa616685db3bfefba947535b2L1145, and is now true based on https://github.com/apache/airflow/blob/de75b7a2bd7f5bef6a1d09942e0b43c17a3fbb95/airflow/ti_deps/deps/prev_dagrun_dep.py#L75

**Comment by feng-tao** [comment]
lgtm. thanks @dima-asana

**Pull request merged**